lang/py/avro/test/test_schema.py

```diff
@@ -629,6 +641,10 @@ def test_invalid_name(self):
             None,
             None,
         )
+        # A name cannot start with dot."
```

Comment:
name.py line 83 says ` # The empty string may be used as a namespace to indicate the null namespace.`
The spec [says](https://avro.apache.org/docs/1.11.1/specification/#names) the same.

Reply:
A few lines later, the spec says:

> The null namespace may not be used in a dot-separated sequence of names.

Reply:
This is currently under discussion on the mailing list -- I'm working through some issues with Java on `.MyRecord` style qualified names to see what the impact would be, but it's currently not permitted by the spec.

Reply:
> A few lines later, the spec says:
> 
> > The null namespace may not be used in a dot-separated sequence of names.

This is consistent with the canonical form detailed here: https://avro.apache.org/docs/1.11.1/specification/_print/#transforming-into-parsing-canonical-form

One would assume that a schema and its canonical form must have the same validation result?

Reply:
> The null namespace may not be used in a dot-separated sequence of names.

I think the idea behind this text is that `"name": ".some.ns.Abc"` is not allowed. I.e. one cannot use the `null` namespace when there is more in the namespace part of the name.

Reply:
To be very clear : `.Abc` is not currently a supported fullname.   Best practice strongly suggests that you don't use an unqualified name in a nested named schema.  Today, you're likely to run into undefined and unexpected behaviour if you you do!

That can of worms needs to be solved but not necessarily in this PR...

Reply:
> That can of worms needs to be solved but not necessarily in this PR...

Agreed!